### PR TITLE
CLIENTS: Remove File By Id

### LINKED ISSUE #339: CLIENTS: Remove File By Id
# The Ask
We need to develop a Client Method that Removes a File by Id by leveraging Foundation Service.

# How to Complete this Task
Here's some steps to complete this deliverable.

- Learn what Exposers are [here](https://github.com/hassanhabib/The-Standard/blob/master/3.%20Exposers/3.%20Exposers.md#3-exposers)
- Watch how The Standard Community developed a similar Client for OpenAI [here](https://youtu.be/OvU6Tate4iI?t=6089)
- If you have any questions - ask The Standard Community [here](https://discord.com/invite/vdPZ7hS52X)
- You cannot start working on this issue until #338  is completed and closed

# How to claim this issue?
- Post on this issue a comment that says: "I claim this issue"
- You will be given 24 hours to complete the issue - if no PR is opened within 24 hours someone else will claim it
- If you have an open PR but is inactive for more than 24 hours from the last review comment we will shutdown your PR

### PR DESCRIPTION
**Pull request by GulchexrasGithub**
Closes #339

### CODE REVIEW

File: Standard.AI.OpenAI/Clients/Files/FilesClient.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            {
+                throw new FileDependencyException(
+                    fileDependencyException.InnerException as Xeption);
+            }
+            catch (FileServiceException fileServiceException)
+            {
+                throw new FileServiceException(
+                    fileServiceException.InnerException as Xeption);
+            }
+        }
```
Comment (Gummimundur) [inline Standard.AI.OpenAI/Clients/Files/FilesClient.cs:46]:
```suggestion
        public async ValueTask<File> RemoveFileByIdAsync(string fileId)
        {
            try
            {
                return await this.fileService.RemoveFileByIdAsync(fileId);
            }
            catch (FileValidationException fileValidationException)
            {
                throw new FileClientValidationException(
                    fileValidationException.InnerException as Xeption);
            }
            catch (FileDependencyValidationException fileDependencyValidationException)
            {
                throw new FileClientValidationException(
                    fileDependencyValidationException.InnerException as Xeption);
            }
            catch (FileDependencyException fileDependencyException)
            {
                throw new FileClientDependencyException(
                    fileDependencyException.InnerException as Xeption);
            }
            catch (FileServiceException fileServiceException)
            {
                throw new FileClientServiceException(
                    fileServiceException.InnerException as Xeption);
            }
        }
```

This client will need to have its own exceptions defined `Standard.AI.OpenAI/Models/Clients/Files/Exceptions/`

**Comment by Gummimundur** [comment]
@GulchexrasGithub  
This client will need to be added to `OpenAIClient`.

Note: There were some significant merge conflicts in #349, that required the `RemoveFileByIdAsync` to be moved to `AIFileService` and `FileService` to be deleted. Therefore this PR might have to wait until #349 is merged.

**Comment by GulchexrasGithub** [comment]
> @GulchexrasGithub This client will need to be added to `OpenAIClient`.
> 
> Note: There were some significant merge conflicts in #349, that required the `RemoveFileByIdAsync` to be moved to `AIFileService` and `FileService` to be deleted. Therefore this PR might have to wait until #349 is merged.

Okay, I'll wait.

**Comment by hassanhabib** [comment]
Orchestration Service implementation isn't ready for this yet.

**Comment by hassanhabib** [comment]
Shutting down due to inactivity

**Pull request closed without merging**